packages/shared/src/multisig.proxy.ts

```diff
@@ -0,0 +1,726 @@
+  )
+
+  // Funds the multisig account to execute the call
+  const extraFunds = 1e10
+  await setupBalances(client, [{ address: multisigAddress, amount: transferAmount + extraFunds }])
+
+  // Dave should have no reserved funds yet.
+  let daveReservedFunds = await getReservedFunds(client, dave.address)
+  expect(daveReservedFunds, 'Dave should have no reserved funds').toBe(0)
+
+  // Dave adds the multisig as his proxy.
+  const addProxyTx = client.api.tx.proxy.addProxy(multisigAddress, 'Any', 0)
+  await sendTransaction(addProxyTx.signAsync(dave))
+
+  await client.dev.newBlock()
+
+  // Check that Dave has had funds reserved for the proxy.
+  daveReservedFunds = await getReservedFunds(client, dave.address)
+  expect(daveReservedFunds, 'Dave should have reserved funds').toBe(await getProxyCosts(client, 1))
+
+  // Check that Charlie has no free funds.
+  let charlieFreeFunds = await getFreeFunds(client, charlie.address)
+  const daveOldFreeFunds = await getFreeFunds(client, dave.address)
+  expect(charlieFreeFunds, 'Charlie should have no free funds').toBe(0)
+
+  // Approve the multisig call. This is the final approval, so `multisig.asMulti` is used.
+  otherSignatories = [alice.address, charlie.address].sort()
+
+  const finalApprovalTx = client.api.tx.multisig.asMulti(
+    threshold,
+    otherSignatories,
+    {
+      height: currBlockNumber + 1,
+      index: multisigExtrinsicIndex,
+    },
+    proxyTx.method.toHex(),
+    maxWeight,
+  )
+
+  const finalApprovalEvents = await sendTransaction(finalApprovalTx.signAsync(bob))
+
+  await client.dev.newBlock()
+
+  await checkEvents(finalApprovalEvents, 'multisig')
+    .redact({
+      redactKeys: /height/,
+    })
+    .toMatchSnapshot('events when Bob approves multisig call')
+
+  // Check that Alice's deposit is gone.
+  aliceReservedFunds = await getReservedFunds(client, alice.address)
+  expect(aliceReservedFunds, "Alice's deposit should have been refunded").toBe(0)
+
+  // Check that Dave has lost funds and Charlie has gained funds.
+  const daveNewFreeFunds = await getFreeFunds(client, dave.address)
+
+  charlieFreeFunds = await getFreeFunds(client, charlie.address)
+  expect(charlieFreeFunds, 'Charlie should have some free funds after the multisig executes').toBe(transferAmount)
+  expect(daveOldFreeFunds - daveNewFreeFunds, 'Dave should have lost some free funds after the multisig executes').toBe(
+    transferAmount,
+  )
+}
+
+/**
+ * Test basic multisig-with-pure-proxy-multisig creation and execution.
+ *
+ * 1. Charlie creates a pure proxy which will later change ownership to the secondary multisig
+ *  - This is needed since the proxy must already exist when creating the secondary multisig
+ * 2. Alice creates a 2-of-3 multisig operation with Bob and Charlies pure proxy as other signatories
+ *  - The operation is to send funds from the primary multisig to Dave
+ *
+ * 1. Alice creates a 2-of-3 multisig operation with Bob and Charlie as other signatories
```

Comment:
Could this perhaps be leftover from earlier iterations of the test? That the numbering starts over, and that an `Eve` appears in the test makes me think so.